### PR TITLE
Extract dep on jjwt-api plugin

### PR DESCRIPTION
**Pull request by jglick**
Extracts the dep added in #269 to its own plugin.

Subsumes #365.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -56,21 +51,9 @@
...
-            <artifactId>jjwt-impl</artifactId>
-            <version>${jjwt.version}</version>
-            <scope>runtime</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.jsonwebtoken</groupId>
-            <artifactId>jjwt-jackson</artifactId>
-            <version>${jjwt.version}</version>
-            <scope>runtime</scope>
+            <version>0.11.2-999999-SNAPSHOT</version>
```
Comment (jglick) [inline pom.xml:56]: pending https://issues.jenkins.io/browse/HOSTING-1060

**Review by dwnusbaum — APPROVED** [review]

**Comment by jglick** [comment]
waiting on https://github.com/jenkins-infra/repository-permissions-updater/pull/1800

**Comment by jglick** [comment]
Did an interactive sanity check in `java -jar` mode (not `hpi:run`) to make sure there are no class loading issues. Was able to define App credentials and use them to scan an org.

**Review by bitwiseman — APPROVED** [review]

File: pom.xml [hunk 2]
```diff
@@ -56,21 +51,9 @@
...
-            <artifactId>jjwt-impl</artifactId>
-            <version>${jjwt.version}</version>
-            <scope>runtime</scope>
-        </dependency>
-        <dependency>
-            <groupId>io.jsonwebtoken</groupId>
-            <artifactId>jjwt-jackson</artifactId>
-            <version>${jjwt.version}</version>
-            <scope>runtime</scope>
+            <version>0.11.2-3.3ae26ccdac4e</version>
```
Comment (bitwiseman) [inline pom.xml:56]: @jglick  
So, this is the final released version?  Just double checking.

Comment (jglick) [reply pom.xml:56]: Yes! See JEP-229.

**Pull request merged**